src/khoj/processor/notion/notion_to_jsonl.py

```diff
@@ -0,0 +1,126 @@
+        current_entries = []
+
+        # Get all pages
+        with timer("Getting all pages via search endpoint", logger=logger):
+            responses = []
+
+            while True:
+                result = self.session.post(
+                    "https://api.notion.com/v1/search",
+                    json={"page_size": 100},
+                ).json()
+                responses.append(result)
+                if result["has_more"] == False:
+                    break
+                else:
+                    self.session.params = {"start_cursor": responses[-1]["next_cursor"]}
+
+        for response in responses:
+            with timer("Processing response", logger=logger):
+                pages_or_databases = response["results"]
+
+                # Get all pages content
+                for p_or_d in pages_or_databases:
+                    with timer(f"Processing {p_or_d['object']} {p_or_d['id']}", logger=logger):
+                        if p_or_d["object"] == "database":
+                            # TODO: Handle databases
+                            continue
+                        elif p_or_d["object"] == "page":
+                            page_id = p_or_d["id"]
+                            title, content = self.get_page_content(page_id)
+                            raw_content = ""
+                            curr_heading = ""
+                            for block in content["results"]:
+                                block_type = block.get("type", None)
+                                if block_type == None:
+                                    continue
+                                block_data = block[block_type]
+                                raw_content = ""
+                                if block_data.get("rich_text", None) and len(block_data["rich_text"]) > 0:
+                                    if block_type in ["heading_1", "heading_2", "heading_3"]:
+                                        if raw_content != "":
+                                            current_entries.append(
+                                                Entry(
+                                                    compiled=raw_content,
+                                                    raw=raw_content,
+                                                    heading=title,
+                                                    file=p_or_d["url"],
+                                                )
+                                            )
+                                        curr_heading = block_data["rich_text"][0]["plain_text"]
+                                    if curr_heading != "":
+                                        raw_content = curr_heading + "\n"
+                                    for text in block_data["rich_text"]:
+                                        raw_content += text["plain_text"]
+                                        raw_content += "\n"
+                                if raw_content != "":
```

Comment:
Also it'd be great if we can add a comment here, specifying that this should add any last heading into the current_entries list
```suggestion
                                // Append the last entry, if any, from Notion for indexing
                                if raw_content != "":
```